### ISSUE TITLE
TypeError: Function.prototype.bind.apply(...) is not a constructor for mm.foundation.modal

### ISSUE THREAD

**Issue by ryankc33** [circlingthesun/angular-foundation-6#21]
Received the following error when trying to include the mm.foundation.modal module:

````
application-2d1cebd….js:4632Uncaught Error: [$injector:modulerr] Failed to instantiate module App due to:
Error: [$injector:modulerr] Failed to instantiate module mm.foundation.modal due to:
TypeError: Function.prototype.bind.apply(...) is not a constructor
````

the problem was traced back to line 362 of src/modal/modal.js. 
`.provider('$modal',  () => {`

Changing the line above to:
`.provider('$modal', function () {`
worked and the module no longer throws the error.  

It could be related to this:
angular/angular.js#14814

**Comment by circlingthesun** [comment]
This is because the original source uses ES6. This gets bundled and compiled down to ES5 along with the other modules into angular-foundation.js. If you have a build pipeline you'd want to pipe the original source files through a transpiler like babel first.

**Comment by circlingthesun** [comment]
Mmmm, I might have jumped the gun there. I've not encountered this issue before. If provider() expects a newable function, that's likely a bug.

**Comment by circlingthesun** [comment]
That provider actually made no sense. I've reshuffled the code and released v0.10.7. Let me know if this fixes things.

**Comment by ryankc33** [comment]
@circlingthesun Yup the error's gone. Closing the issue.
